scalafix-reflect/src/main/scala-3/scalafix/internal/reflect/RuleCompiler.scala

```diff
@@ -0,0 +1,71 @@
+package scalafix.internal.reflect
+
+import scalafix.internal.reflect.exception.CompilationException
+
+import java.io.File
+import java.nio.file.Paths
+import scala.meta.io.AbsolutePath
+
+import dotty.tools.dotc.Compiler
+import dotty.tools.dotc.Run
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.core.Contexts.NoContext.given_Context
+import dotty.tools.dotc.reporting.StoreReporter
+
+import scala.reflect.io.VirtualDirectory
+import scala.reflect.io.AbstractFile
+import scala.reflect.internal.util.AbstractFileClassLoader
+
+import metaconfig.Configured
+import metaconfig.Input
+
+class RuleCompiler(
+    classpath: String,
+    target: AbstractFile = new VirtualDirectory("(memory)", None)
+) {
+  // we are not configuring any settings!
+  // private val settings = new Settings()
+  // settings.deprecation.value = true // enable detailed deprecation warnings
+  // settings.unchecked.value = true // enable detailed unchecked warnings
+  // settings.outputDirs.setSingleOutput(target)
+  // settings.classpath.value = classpath
```

Comment:
irrelevant with dotty's API?

Reply:
I've found a way to provide such settings, will remove the commented lines once we think RuleCompiler is good enough